src/python/pants/backend/docker/goals/package_image.py

```diff
@@ -512,8 +512,11 @@ def parse_image_id_from_docker_build_output(*outputs: bytes) -> str:
                 r"(writing image (?P<digest>sha256:\S+) done)",
                 # Docker output.
                 r"(Successfully built (?P<short_id>\S+))",
+                # Podman output
+                r"^(?P<podman_long_id>\S{64})$",
```

Comment:
I'm not quite sure about this bit. It's relying on three different binaries working differently to infer the expected result.

Reply:
Yea, ideally we could keep some information around about which binary we found.. (or rely on the found path)

otherwise, can we grep multiple lines here, as the second best thing?
```suggestion
                r"^Successfully tagged .*\n(?P<podman_long_id>\S{64})$",
```

Reply:
> Yea, ideally we could keep some information around about which binary we found.. (or rely on the found path)
> 
> otherwise, can we grep multiple lines here, as the second best thing?

This is what I originally had, but the code below splits it out and searches line-by-line, not multiline (presumably to give precedence to the last line)

Reply:
Ok, I have added `DockerBinary.is_podman` that is set when podman is enabled & discovered.

I then check for this when parsing the output of `podman build` and do some regex-free checking for a success string.

The previous "blind faith" approach does cover the edge case where someone symlinks `docker -> podman`. The build and image-id would be recognized. This approach leads to `<unknown>`.

I feel this is _slowly_ eeking towards "why don't we use Union rules to support other tools?" but given Podman is meant to be a drop-in replacement, maybe this is all that's needed? Is the likelihood of a third docker-binary-compatible drop-in container-system really that high?

Reply:
yea, true. don't spend too much effort here. We may even want to go in a different direction eventually to use a docker lib to talk directly with the docker API instead of using a cli client. (as we do for the docker environments support, i.e. lean into that work more and build upon it)